### PR TITLE
chore: windows ci

### PR DESCRIPTION
**Pull request by styfle**
Related to #689

### CODE REVIEW

**Comment by guybedford** [comment]
Perhaps try updating the version of `node-gyp` here?

**Comment by styfle** [comment]
@guybedford Bumping node-gyp helped but now jest doesn't run. Seems like we need to expose GC and make it windows compatible in order to run the tests.

File: package.json [hunk 1]
```diff
@@ -15,8 +15,8 @@
     "build": "node scripts/build",
     "build-test-binary": "cd test/binary && node-gyp rebuild && cp build/Release/hello.node ../integration/hello.node",
     "codecov": "codecov",
-    "test": "node --expose-gc --max_old_space_size=3072 node_modules/.bin/jest",
-    "test-coverage": "node --expose-gc --max_old_space_size=3072 node_modules/.bin/jest --coverage --globals \"{\\\"coverage\\\":true}\" && codecov",
```
Comment (guybedford) [inline package.json:19]: `--max_old_space_size` is necessary here.

Comment (styfle) [reply package.json:19]: Thanks! I added cross-spawn and that got past the memory issue, but now jest doesn't find any tests on Windows 🤔

**Comment by guybedford** [comment]
Seems like it might have something to do with the first `?` in the configuration matcher - https://github.com/vercel/ncc/blob/master/jest.config.js#L5.

CI reports: `testMatch: D:/a/ncc/ncc/test\?(*.)+(spec|test).js?(x) - 0 matches`

While the matcher configuration has: `testMatch: ["<rootDir>/test/?(*.)+(spec|test).js?(x)"]`

Try simplifying the rule to just:

```
testMatch: ["<rootDir>/test/*.test.js"]
```

**Comment by styfle** [comment]
Good call!

Looks like some of the tests are failing because windows emits `\r\n` instead of the expected `\n` so thats not a big issue.

However, there are a couple that look like actual bugs: https://github.com/vercel/ncc/pull/691/checks?check_run_id=2333639573#step:9:318

**Comment by styfle** [comment]
Fixed in #896

**Pull request closed without merging**